pennylane/devices/default_tensor.py

```diff
@@ -618,7 +634,25 @@ def simulate(self, circuit: QuantumScript) -> Result:
 
         self._initialize_quimb_circuit(wires)
 
-        for op in circuit.operations:
+        for i, op in enumerate(circuit.operations):
+            if i == 0 and isinstance(op, qml.BasisState):
+                self._quimb_circuit = qtn.CircuitMPS(
+                    psi0=self._initial_mps(
+                        op.wires, basis_state="".join(str(int(b)) for b in op.parameters[0])
+                    ),
+                    max_bond=self._max_bond_dim,
+                    gate_contract=self._contract,
+                    cutoff=self._cutoff,
+                )
+                continue
+            if i == 0 and isinstance(op, qml.StatePrep):
+                self._quimb_circuit = qtn.CircuitMPS(
+                    psi0=qtn.MatrixProductState.from_dense(op.parameters[0]),
+                    max_bond=self._max_bond_dim,
+                    gate_contract=self._contract,
+                    cutoff=self._cutoff,
+                )
+                continue
```

Comment:
If the user provides `method="tn"`, it seems that the `quimb` instance is created here using the `CircuitMPS` class regardless of the chosen method

Reply:
Fixed calling `_initial_quimb_circuit`.